Update compilation documentation + Fix debugging for Microsoft OS

nmellado opened the pull request:
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

Be aware that the PR request cannot be accepted if it doesn't pass the Continuous Integration tests.


* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Doc update


* **What is the current behavior?** (You can also link to an open issue here)
Current build instruction described in documentation for MSVC+ninja do not work on VS 2019.


* **What is the new behavior (if this is a feature change)?**
Update doc


* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
no


* **Other information**:

hoshiryu commented:
Could you provide instructions about how to debug a main-app plugin? 
I mean all the configuration/compilation/install procedure and what is needed beforehand from the windows dev toolkit.
Thanks

nmellado commented:
Ok will do it.

nmellado commented:
@hoshiryu the following is working on my side (using this PR), could you try it ?
 - Compile Radium-Engine in debug mode
 - Install it
 - Compile Radium-Apps in debug mode
 - Start `main-app.exe (install)`
![image](https://user-images.githubusercontent.com/6310221/88028297-beec8480-cb38-11ea-95ef-d8bdb99b2856.png)

dlyr commented:
**CHECK LINEAR HISTORY FAIL**

Please rebase on current up to date master,
without merge commits (e.g. you can squash them).

nmellado commented:
@hoshiryu I fixed a bug preventing to start the applications, and I added the installation of the pdb files in debug mode.
To test, you simply need to recompile and install Radium-Engine (don't need to reconfigure nor to empty the cmake cache).
Then you can recompile your application and should be able to debug: 

![image](https://user-images.githubusercontent.com/6310221/88035782-ef392080-cb42-11ea-9316-8300c6e6293c.png)

MathiasPaulin commented on cmake/radium_setup_functions.cmake:
```diff
@@ -59,6 +59,7 @@ function(configure_cmdline_Radium_app)
         TARGETS ${ARGS_NAME}
         RUNTIME DESTINATION bin
     )
+
```
```suggestion
```

MathiasPaulin commented on cmake/radium_setup_functions.cmake:
```diff
@@ -128,6 +132,7 @@ function(configure_bundled_Radium_app)
         TARGETS ${ARGS_NAME}
         BUNDLE DESTINATION "bin/"
     )
+
```
```suggestion
```

MathiasPaulin commented on doc/basics/compilation.md:
```diff
@@ -167,34 +163,39 @@ To fix them, you need to edit the VS-specific file `CMakeSettings.json`, via *CM
 ### Compilation
 
 Right click on CMakeList.txt > build > all.
+To install, you need to run any installation target, e.g. `HelloRadium.exe (install)`.
```
```suggestion
To install, you need to run any installation target, e.g. `Engine.dll (install)` or to select the menu <Build>/<Install radiumproject>
```

nmellado commented on src/Engine/Renderer/Mesh/Mesh.hpp:
```diff
@@ -138,8 +138,13 @@ class RA_ENGINE_API AttribArrayDisplayable : public Displayable
+                m_displayable->m_isDirty          = true;
+            }
+            else
+                LOG( logDEBUG ) << "Invalid dirty bit notified";
```
```suggestion
                LOG( logDEBUG ) << "Invalid dirty bit notified"; /// \fixme Should never be here
```
